### PR TITLE
promoting global vehicle data/information to Vehicle root branch

### PR DESCRIPTION
**Pull request by chheis**
promoting global vehicle data/information to Vehicle root branch **_using deprecation_**

### removed
Navigation.CurrentLocation.Speed **removed** because doubled with Vehicle.Speed

### moved
complete branch: Navigation.CurrentLocation moved to Vehicle.CurrentLocation
complete branch: Chassis.Trailer moved to Vehicle.Trailer
Vehicle masses moved from Chassis to Vehicle

Signed-off-by: Christian Heissenberger <christian.heissenberger@de.bosch.com>

### CODE REVIEW

File: spec/Vehicle/Vehicle.vspec [hunk 1]
```diff
@@ -295,3 +295,123 @@
...
+#
+# Vehicle Weight and Dimension attributes
+#
+
+- CurrentOverallWeight:
+  datatype: uint16
+  type: sensor
+  value: 0
+  unit: kg
+  description: Current overall Vehicle weight, in kg, including all liquids and full tank of fuel.
```
Comment (erikbosch) [inline spec/Vehicle/Vehicle.vspec:309]: I know that we just moved this item, but is the description mentioning "full tank of fuel" correct? I.e. that the sensor providing this value needs to consider how full the tank is and adjust the value accordingly?

Comment (SebastianSchildt) [reply spec/Vehicle/Vehicle.vspec:309]: My feeling is that "CurrentOverallWeight" is sort of the "best guess" how heavy a vehicle currently is, including everything potentially even cargo and passengers. (how accurate it is/can be depends on the vehicle).
@chheis may correct me 😀

Comment (danielwilms) [reply spec/Vehicle/Vehicle.vspec:309]: I follow @erikbosch. from the description I'd rather think it's a static value, but the type `sensor` indicates, that it's dynamic. Maybe we should clarify that.

**Review by danielwilms — COMMENTED** [review]
I'm undecided if you should have the deprecation just on the branch or on every leaf as well. That we should discuss.

File: spec/Chassis/Chassis.vspec [hunk 2]
```diff
@@ -220,8 +226,10 @@
 - Trailer:
   type: branch
   description: Trailer signals
+  deprecation: @2.1 moved branch to Vehicle.Trailer
 
 - Trailer.Connected:
   datatype: boolean
   type: sensor
   description: Signal indicating if trailer is connected or not.
+  deprecation: @2.1 moved to Vehicle.Trailer.Width
```
Comment (danielwilms) [inline spec/Chassis/Chassis.vspec:235]: looks like a copy paste error.

**Comment by gunnarx** [comment]
I think each signal should have its own deprecation notice.  Even if it is a bit more work, it simplifies processing and I suppose in the future there should hopefullly not be too many of these major moves of complete branches...

**Comment by gunnarx** [comment]
@danielwilms Check the build failure.  Perhaps this usage of @ is not valid YAML

File: spec/Vehicle/Vehicle.vspec (hunk 1 above)
Comment (chheis) [reply spec/Vehicle/Vehicle.vspec]: I changed the description according to the discussion

File: spec/Chassis/Chassis.vspec (hunk 2 above)
Comment (chheis) [reply spec/Chassis/Chassis.vspec]: thanks, corrected

**Comment by gunnarx** [comment]
There is a parsing error when encountering the @ symbol on the deprecation line.
Also https://yamlvalidator.com seems to complain about the format?

**Pull request merged**